x-pack/plugins/maps/public/shared/layers/sources/client_file_source/geojson_file_source.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AbstractVectorSource } from '../vector_source';
+import React from 'react';
+import { GEOJSON_FILE } from '../../../../../common/constants';
+import { ClientFileCreateSourceEditor } from './create_client_file_source_editor';
+import { ESSearchSource } from '../es_search_source';
+import uuid from 'uuid/v4';
+import _ from 'lodash';
+
+export class GeojsonFileSource extends AbstractVectorSource {
+
+  static type = GEOJSON_FILE;
+  static title = 'Import GeoJSON Vector File';
+  static description = 'Upload a GeoJSON file & Index in Elasticsearch';
+  static icon = 'emsApp';
```

Comment:
We probably need a new icon for this cc @cchaos 

I think `importAction` (https://elastic.github.io/eui/#/display/icons) probably works better here

Reply:
I agree that the `importAction` icon is a better fit here and distinguishes the capability of this source vs. the others. I've updated it!